### PR TITLE
update formatDate function

### PR DESCRIPTION
**Pull request by inclinedmind**
add korean, and use local storage locale as backup

Screen Shot 2023-07-21 at 8 36 07 PM

### CODE REVIEW

File: packages/utils/src/formatDate.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { format } from 'date-fns';
-import { enUS, es } from 'date-fns/locale';
+import { enUS, es, ko } from 'date-fns/locale';
```
Comment (codewiththeflowxyz) [inline packages/utils/src/formatDate.ts:2]: I think this will be very difficult to maintain when we include all the locales we support. Maybe better if we try to load the locale dynamically. 

// Get the locales
import * as locales from 'date-fns/locale';

const[code, region] = locale.split('-')
const locale = locales[`${code}-${region}`] ?? locales[code] ?? locales.enUS

Comment (inclinedmind) [reply packages/utils/src/formatDate.ts]: Ah good idea, I looked at the list of locales used by date-fns (https://github.com/date-fns/date-fns/tree/main/src/locale) and it seems like this strategy works well

File: packages/utils/src/formatDate.ts [hunk 2]
```diff
@@ -1,22 +1,15 @@
...
-    case 'es':
-    case 'es-AR':
-    case 'es-ES':
-      return es;
-    default:
-      return enUS;
-  }
+const getDateLocale = (locale: string): Locale => {
+  const [code, region] = locale.split('-');
+  // @ts-ignore
```
Comment (codewiththeflowxyz) [inline packages/utils/src/formatDate.ts:8]: Why is this needed?

**Review by codewiththeflowxyz — APPROVED** [review]
Looking good :)

File: packages/utils/src/formatDate.ts (hunk 2 above)
Comment (inclinedmind) [reply packages/utils/src/formatDate.ts]: added a couple `as keyof typeof locales` and removed the @ts-ignore
